### DISCUSSION TITLE
Does VS Code support voice command?

### DISCUSSION THREAD

**Discussion by sadmankiba** [microsoft/vscode-discussions#79] [Extension Development QnA] [answered]
Does VS Code support voice command? With any extension or any third-party tool? If not, is there any development plan to do so?

Some useful features might be directory navigation, speech-to-text, search text or file with speech etc.

**Comment by bwateratmsft** [comment]
VSCode, open PodBayDoor.ts

\> I'm sorry Dave, I'm afraid I can't do that.

![image](https://user-images.githubusercontent.com/36966225/181764978-726baeeb-a432-4ee6-a981-f6b9222d1aa8.png)

Sorry, I couldn't resist :smile:. I do think this is a neat idea though, it could help with accessibility significantly.

**Reply by sadmankiba** [reply]
Looks like [Serenade](https://serenade.ai/) has an extension for this. Not sure if it talks back :smile:

**Comment by isidorn** [comment]
This is currently not supported by VS Code. But extensions could do this.
There is already a discussion about this here https://github.com/microsoft/vscode/issues/40976
A I know there are some extensions trying to do this.

**Comment by mika76** [comment, marked as answer]
Noticed this while searching for some info - there is a new Microsoft extension called [vscode speech](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-speech) - not sure about commands though.
